src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -87,83 +87,71 @@ public static void ToCharsBuffer(byte value, Span<char> buffer, int startingInde
         }
 
 #if SYSTEM_PRIVATE_CORELIB
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static (Vector128<byte>, Vector128<byte>) AsciiToHexVector128(Vector128<byte> src, Vector128<byte> hexMap)
+        {
+            Debug.Assert(Ssse3.IsSupported || AdvSimd.Arm64.IsSupported);
```

Comment:
Same as before, I believe this pattern is problematic for crossgen and we expect a path that throws.

Reply:
@tannergooding I don't expect this function to have an R2R version 🤷 why is it needed?

Reply:
Anything in corelib gets some special assumptions in how IsSupported checks work

The rules for CG1: https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/botr/vectors-and-intrinsics.md#crossgen1-model-of-hardware-intrinsic-usage

and for CG2: https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/botr/vectors-and-intrinsics.md#crossgen2-model-of-hardware-intrinsic-usage

It looks like the problematic scenario I'm remembering might have been CG1 only, but I didn't re-read the rules in depth.

Reply:
I double checked via `R2RDump.exe` that these functions don't present in the R2R'd corelib. Also, all the callsites where they were inlined are prejitted just fine, e.g. here is `Utf8Formatter.TryFormat`:
```asm
; bool System.Buffers.Text.Utf8Formatter.TryFormat(System.Guid, System.Span`1<byte>, ref int, System.Buffers.StandardFormat)
2c78d0: 48 83 ec 38             sub     rsp, 56
                                UWOP_ALLOC_SMALL 56                                0

2c78d4: 4c 89 4c 24 58          mov     qword ptr [rsp + 88], r9
2c78d9: 4c 8b 0a                mov     r9, qword ptr [rdx]
2c78dc: 8b 52 08                mov     edx, dword ptr [rdx + 8]
2c78df: 0f b6 44 24 58          movzx   eax, byte ptr [rsp + 88]
2c78e4: 85 c0                   test    eax, eax
2c78e6: 75 0c                   jne     0x2c78f4
2c78e8: 80 7c 24 59 00          cmp     byte ptr [rsp + 89], 0
2c78ed: 75 05                   jne     0x2c78f4
2c78ef: b8 44 00 00 00          mov     eax, 68
2c78f4: 83 f8 44                cmp     eax, 68
2c78f7: 77 16                   ja      0x2c790f
2c78f9: 83 f8 42                cmp     eax, 66
2c78fc: 74 27                   je      0x2c7925
2c78fe: 83 f8 44                cmp     eax, 68
2c7901: 0f 85 5c 01 00 00       jne     0x2c7a63
2c7907: 41 ba 24 00 00 80       mov     r10d, 2147483684
2c790d: eb 24                   jmp     0x2c7933
2c790f: 83 f8 4e                cmp     eax, 78
2c7912: 74 19                   je      0x2c792d
2c7914: 83 f8 50                cmp     eax, 80
2c7917: 0f 85 46 01 00 00       jne     0x2c7a63
2c791d: 41 ba 26 28 29 80       mov     r10d, 2150180902
2c7923: eb 0e                   jmp     0x2c7933
2c7925: 41 ba 26 7b 7d 80       mov     r10d, 2155707174
2c792b: eb 06                   jmp     0x2c7933
2c792d: 41 ba 20 00 00 00       mov     r10d, 32
2c7933: 41 0f b6 c2             movzx   eax, r10b
2c7937: 3b c2                   cmp     eax, edx
2c7939: 7e 0a                   jle     0x2c7945
2c793b: 33 c0                   xor     eax, eax
2c793d: 41 89 00                mov     dword ptr [r8], eax
2c7940: 48 83 c4 38             add     rsp, 56
2c7944: c3                      ret
2c7945: 41 0f b6 c2             movzx   eax, r10b
2c7949: 41 89 00                mov     dword ptr [r8], eax
2c794c: 41 c1 fa 08             sar     r10d, 8
2c7950: 45 84 d2                test    r10b, r10b
2c7953: 74 17                   je      0x2c796c
2c7955: 85 d2                   test    edx, edx
2c7957: 0f 84 19 01 00 00       je      0x2c7a76
2c795d: 45 88 11                mov     byte ptr [r9], r10b
2c7960: 83 fa 01                cmp     edx, 1
2c7963: 0f 82 06 01 00 00       jb      0x2c7a6f
2c7969: 49 ff c1                inc     r9
2c796c: 41 c1 fa 08             sar     r10d, 8
2c7970: 0f 10 01                movups  xmm0, xmmword ptr [rcx]
2c7973: 0f 11 44 24 28          movups  xmmword ptr [rsp + 40], xmm0
2c7978: 48 8d 05 41 ab eb ff    lea     rax, [0x1824c0]
2c797f: 0f 10 08                movups  xmm1, xmmword ptr [rax]
2c7982: 0f 28 d0                movaps  xmm2, xmm0
2c7985: 66 0f 73 d2 04          psrlq   xmm2, 4
2c798a: 0f 28 da                movaps  xmm3, xmm2
2c798d: 66 0f 60 d8             punpcklbw xmm3, xmm0
2c7991: 66 0f 68 d0             punpckhbw xmm2, xmm0
2c7995: 0f 10 05 b4 c8 6f 00    movups  xmm0, xmmword ptr [0x9c4250]
2c799c: 66 0f db d8             pand    xmm3, xmm0
2c79a0: 0f 28 e1                movaps  xmm4, xmm1
2c79a3: 66 0f 38 00 e3          pshufb  xmm4, xmm3
2c79a8: 66 0f db c2             pand    xmm0, xmm2
2c79ac: 66 0f 38 00 c8          pshufb  xmm1, xmm0
2c79b1: 0f 10 05 a8 c8 6f 00    movups  xmm0, xmmword ptr [0x9c4260]
2c79b8: 66 0f 38 00 e0          pshufb  xmm4, xmm0
2c79bd: 45 85 d2                test    r10d, r10d
2c79c0: 7d 53                   jge     0x2c7a15
2c79c2: 0f 10 05 a7 c8 6f 00    movups  xmm0, xmmword ptr [0x9c4270]
2c79c9: 0f 28 d4                movaps  xmm2, xmm4
2c79cc: 66 0f 38 00 d0          pshufb  xmm2, xmm0
2c79d1: 0f 10 05 a8 c8 6f 00    movups  xmm0, xmmword ptr [0x9c4280]
2c79d8: 0f 28 d9                movaps  xmm3, xmm1
2c79db: 66 0f 38 00 d8          pshufb  xmm3, xmm0
2c79e0: 0f 10 05 a9 c8 6f 00    movups  xmm0, xmmword ptr [0x9c4290]
2c79e7: 66 0f 38 00 e0          pshufb  xmm4, xmm0
2c79ec: 0f 10 05 ad c8 6f 00    movups  xmm0, xmmword ptr [0x9c42a0]
2c79f3: 66 0f 38 00 c8          pshufb  xmm1, xmm0
2c79f8: 66 0f eb cc             por     xmm1, xmm4
2c79fc: 0f 10 05 ad c8 6f 00    movups  xmm0, xmmword ptr [0x9c42b0]
2c7a03: 0f 10 25 b6 c8 6f 00    movups  xmm4, xmmword ptr [0x9c42c0]
2c7a0a: 66 0f 38 00 c4          pshufb  xmm0, xmm4
2c7a0f: 66 0f eb c1             por     xmm0, xmm1
2c7a13: eb 09                   jmp     0x2c7a1e
2c7a15: 0f 28 d4                movaps  xmm2, xmm4
2c7a18: 0f 28 d9                movaps  xmm3, xmm1
2c7a1b: 0f 57 c0                xorps   xmm0, xmm0
2c7a1e: 49 8b c1                mov     rax, r9
2c7a21: 45 85 d2                test    r10d, r10d
2c7a24: 7d 0d                   jge     0x2c7a33
2c7a26: 0f 11 10                movups  xmmword ptr [rax], xmm2
2c7a29: 0f 11 58 14             movups  xmmword ptr [rax + 20], xmm3
2c7a2d: 0f 11 40 08             movups  xmmword ptr [rax + 8], xmm0
2c7a31: eb 07                   jmp     0x2c7a3a
2c7a33: 0f 11 10                movups  xmmword ptr [rax], xmm2
2c7a36: 0f 11 58 10             movups  xmmword ptr [rax + 16], xmm3
2c7a3a: 45 84 d2                test    r10b, r10b
2c7a3d: 74 1a                   je      0x2c7a59
2c7a3f: 45 85 d2                test    r10d, r10d
2c7a42: 7c 08                   jl      0x2c7a4c
2c7a44: 41 b8 20 00 00 00       mov     r8d, 32
2c7a4a: eb 06                   jmp     0x2c7a52
2c7a4c: 41 b8 24 00 00 00       mov     r8d, 36
2c7a52: 49 63 d0                movsxd  rdx, r8d
2c7a55: 44 88 14 10             mov     byte ptr [rax + rdx], r10b
2c7a59: b8 01 00 00 00          mov     eax, 1
2c7a5e: 48 83 c4 38             add     rsp, 56
2c7a62: c3                      ret
2c7a63: 33 c0                   xor     eax, eax
2c7a65: 41 89 00                mov     dword ptr [r8], eax
2c7a68: ff 15 82 67 84 00       call    qword ptr [0xb0e1f0]  // void System.ThrowHelper.ThrowFormatException_BadFormatSpecifier() (METHOD_ENTRY_DEF_TOKEN)
2c7a6e: cc                      int3
2c7a6f: ff 15 83 65 84 00       call    qword ptr [0xb0dff8]  // void System.ThrowHelper.ThrowArgumentOutOfRangeException() (METHOD_ENTRY_DEF_TOKEN)
2c7a75: cc                      int3
2c7a76: ff 15 f4 5f 83 00       call    qword ptr [0xafda70]  // RNG_CHK_FAIL (HELPER)
2c7a7c: cc                      int3
```